utilities/test_suite/HIP/Tensor_hip.cpp

```diff
@@ -672,6 +676,21 @@ int main(int argc, char **argv)
 
                     break;
                 }
+                case 10:
+                {
+                    testCaseName = "fog";
+
+                    for (i = 0; i < batchSize; i++)
+                        gammaTensor[i] = 0;
+
+                    startWallTime = omp_get_wtime();
+                    if (inputBitDepth == 0 || inputBitDepth == 1 || inputBitDepth == 2 || inputBitDepth == 5)
+                        rppt_fog_gpu(d_input, srcDescPtr, d_output, dstDescPtr, gammaTensor, roiTensorPtrSrc, roiTypeSrc, handle);
+                    else
+                        missingFuncFlag = 1;
+
+                    break;
+                }
```

Comment:
where are you freeing the allocated memory for gammaTensor?

You need to free the memory at the end
https://github.com/ROCm/rpp/blob/develop/utilities/test_suite/HIP/Tensor_hip.cpp#L1688

if (gammaTensor != nullptr)
    hipHostFree(gammaTensor);